### PR TITLE
Fix vim CVE in Alibi images

### PR DESCRIPTION
**Pull request by adriangonz**
**What this PR does / why we need it**:
Fix a CVE introduced by Vim in the `seldonio/alibiexplainer` and `seldonio/alibi-detect-server` images.

Also, this PR:
- Moves Conda's base to UBI9 minimal
- Pins Conda to 1.13.0
- Works around an issue building PyStan (one of the prophet's deps)

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
To ensure we are able to rebuild the image, this PR also pins the version of Poetry to `1.1.15`

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **adriangonz**
You can assign the PR to them by writing `/assign @adriangonz` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
@adriangonz: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | e4785d741eac0caf93abaaa07b1b144c8b043d80 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4343-integration-9xq9s) | `/test integration`
notebooks | e4785d741eac0caf93abaaa07b1b144c8b043d80 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-4343-notebooks-6zg6g) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by axsaucedo** [comment]
Just for context, the security tests in the CI can only run when they are a branch part of the SeldonIO repo, so forks cannot trigger as they cannot access the Snyk token - for this if you want to test it you can do it with `act` locally

**Comment by axsaucedo** [comment]
@adriangonz PR looks good - are there any changes required for the artifacts that we'd need to do for the integration tests given the poetry install changed? We only have one test so may be easier to run it locally to validate working instead of the full integration tests.

/test integration

**Comment by adriangonz** [comment]
Hey @axsaucedo ,

The changes ensure that the same version of Poetry gets installed (as in, compared to 1.14.1), so I don't think we need any changes to the artefacts. 

I'll build the images locally and kick off the integration tests locally, as well as the security tests :+1:

**Comment by seldondev** [comment]
@adriangonz: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | e4785d741eac0caf93abaaa07b1b144c8b043d80 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-4343-notebooks-6zg6g) | `/test notebooks`
integration | f9206283aab6e8c353d962a3f3476b335a6aa82d | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4343-integration-t7d5r) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: components/alibi-detect-server/Dockerfile
Comment (RafalSkolasinski) [inline components/alibi-detect-server/Dockerfile:20]: nit + probably personal preference: either I'd keep them in one line or broke down to one item per line

File: components/alibi-detect-server/Dockerfile
Comment (RafalSkolasinski) [inline components/alibi-detect-server/Dockerfile:33]: So the new thing is opting for `conda-forge` channel here - I wonder what is the exact reasoning for it? Could it be that this is what relates to the need for linking `libstdc`?

File: components/alibi-detect-server/Dockerfile
Comment (RafalSkolasinski) [inline components/alibi-detect-server/Dockerfile:48]: Are we plan to move to poetry 1.2.x for 1.15 or we do it post 1.15?

File: components/alibi-detect-server/Dockerfile
Comment (RafalSkolasinski) [inline components/alibi-detect-server/Dockerfile:64]: Are we sure we wanna skip the line with security updates?

File: components/alibi-explain-server/Dockerfile
Comment (RafalSkolasinski) [inline components/alibi-explain-server/Dockerfile:18]: if we opted for `-c conda-forge` in alibi detect image why not here?

File: components/alibi-explain-server/Dockerfile
Comment (RafalSkolasinski) [inline components/alibi-explain-server/Dockerfile:58]: same question as on other image

**Review by RafalSkolasinski — CHANGES_REQUESTED** [review]
LGTM, just some minor questions. The one about:
- `-c conda-forge`
- removal of security update line
applies to all images but I wanted to avoid repetition in the in-line comments

File: components/alibi-detect-server/Dockerfile
Comment (adriangonz) [reply components/alibi-detect-server/Dockerfile:20]: Ok, let me break them down to one item per line

File: components/alibi-detect-server/Dockerfile
Comment (adriangonz) [reply components/alibi-detect-server/Dockerfile:33]: I think I tried both, and neither worked. TBH this is mainly an artefact of trying out diff things to see what worked. 

Happy to revert it back, although I'd be keen to start using `conda-forge` more across our images - mainly to workaround Conda's licensing constrains (haven't checked what's the latest on those though).

File: components/alibi-detect-server/Dockerfile
Comment (adriangonz) [reply components/alibi-detect-server/Dockerfile:64]: Good spot!

I moved them all together to the top of the `Dockerfile`(s), and changed them to `microdnf update -y` (which is the closest equivalent for `microdnf`). 

In some cases, like the one you mentioned, I added it to the interim multi-stage builds, but forgot to add it to the final one as well. I'll double check them to ensure the `microdnf update -y` is present.

File: components/alibi-explain-server/Dockerfile
Comment (adriangonz) [reply components/alibi-explain-server/Dockerfile:18]: Desperation after trying several things out.

I'll add it there though for consistency :+1:

File: components/alibi-detect-server/Dockerfile
Comment (RafalSkolasinski) [reply components/alibi-detect-server/Dockerfile:37]: Fair point, I'd just to just include it in all images for consistency

File: python-builder/Dockerfile
Comment (RafalSkolasinski) [inline python-builder/Dockerfile:42]:
```suggestion
RUN conda install --yes -c conda-forge python=$PYTHON_VERSION conda=$CONDA_VERSION
```

File: wrappers/s2i/python/Dockerfile
Comment (RafalSkolasinski) [inline wrappers/s2i/python/Dockerfile:22]:
```suggestion
    conda install --yes -c conda-forge python=$PYTHON_VERSION conda=$CONDA_VERSION
```

**Review by RafalSkolasinski — CHANGES_REQUESTED** [review]

**Review by RafalSkolasinski — APPROVED** [review]

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[RafalSkolasinski](https://github.com/SeldonIO/seldon-core/pull/4343#pullrequestreview-1151161081 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [RafalSkolasinski]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
